addons/l10n_de/models/datev.py

```diff
@@ -14,3 +16,36 @@ class AccountTax(models.Model):
     _inherit = "account.tax"
 
     l10n_de_datev_code = fields.Char(size=2, help="2 digits code use by Datev")
+
+class AccountInvoiceLine(models.Model):
+    _inherit = "account.invoice.line"
+
+    def write(self, vals):
```

Comment:
do not overwrite `write()` and `create()` but `action_invoice_validate()` instead

Reply:
add a condition to apply this restriction to German companies only

```python
    if self.invoice_id.company_id.country_id.code == 'DE'
```